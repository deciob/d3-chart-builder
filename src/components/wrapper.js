import {
  select
} from "d3-selection";

export default function(config) {
  const svgContainer = d3.select(this);
  const width = config.width;
  const heigh = config.height;
  const margin = confi.margin;

  let svg = svgContainer.select('svg');

  if (svg.empty()) {
    svg = svgContainer.append('svg')
      .attr('width', width + margin.left + margin.right)
      .attr('height', height + margin.top + margin.bottom);
  }

  return svg;
};
