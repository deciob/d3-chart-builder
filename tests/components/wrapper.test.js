import {
  select,
} from 'd3-selection';

import jsdom from 'jsdom';

import wrapper from '../../src/components/wrapper';


test('the svg wrapper component', () => {
  const { JSDOM } = jsdom;
  const state = {
    height: 100,
    width: 100,
  };
  const config = {
    margin: { top: 10, right: 10, bottom: 10, left: 10 },
  };
  const dom = new JSDOM(
    '<div></div>',
  );
  const document = dom.window.document;
  const selection = document.querySelector('div');
  const d3Selection = select(selection);

  wrapper(config, state, d3Selection);

  const outerHtml = dom.window.document.documentElement.outerHTML;
  expect(outerHtml).toMatchSnapshot();
});
