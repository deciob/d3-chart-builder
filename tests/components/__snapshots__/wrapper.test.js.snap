// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the svg wrapper component 1`] = `"<html><head></head><body><div><svg width=\\"120\\" height=\\"120\\"><g transform=\\"translate(10, 10)\\"></g></svg></div></body></html>"`;
